Crashlytics/Crashlytics/FIRCrashlytics.m

```diff
@@ -307,7 +310,14 @@ - (void)processDidCrashDuringPreviousExecution {
 
 #pragma mark - API: Logging
 - (void)log:(NSString *)msg {
-  FIRCLSLog(@"%@", msg);
+  if (!_contextInitPromise) {
+    FIRCLSErrorLog(@"Context has not been inialized when log message: %@", msg);
+    return;
+  }
+  [_contextInitPromise then:^id _Nullable(id _Nullable value) {
+    FIRCLSLog(@"%@", msg);
+    return nil;
+  }];
```

Comment:
2 suggestions:

1. Can we encapsulate this into a method? 

```
waitForContextInit(_contextInitPromise, context: NSString, callback <type>) {

  if (!_contextInitPromise) {
    FIRCLSErrorLog(@"Context has not been inialized for %@", context);
    return;
  }
  [_contextInitPromise then:^id _Nullable(id _Nullable value) {
    FIRCLSLog(@"%@", msg);
    return nil;
}];
}
```

2. Can we make this "Context has not been inialized for" more understandeable for users. Eg. "Crashlytics method called before SDK was initialized: %@", context

Reply:
Added private helper: https://github.com/firebase/firebase-ios-sdk/pull/15356/files#diff-3c7ed577af56820c03d0e93aaff77cc09d547a85837e6e57c66401ae85330033R489